rocketpy/motors/Tank.py

```diff
@@ -0,0 +1,842 @@
+    def mass(self):
+        """Returns the total mass of liquid and gases inside the tank as a
+        function of time.
+
+        Parameters
+        ----------
+        time : float
+            Time in seconds.
+
+        Returns
+        -------
+        Function
+            Mass of the tank as a function of time. Units in kg.
+        """
+        pass
+
+    @abstractmethod
+    def netMassFlowRate(self):
+        """Returns the net mass flow rate of the tank as a function of time.
+        Net mass flow rate is the mass flow rate entering the tank minus the
+        mass flow rate exiting the tank, including liquids and gases. Positive
+        is defined as a net mass flow rate entering the tank.
+
+        Parameters
+        ----------
+        time : float
+            Time in seconds.
+
+        Returns
+        -------
+        Function
+            Net mass flow rate of the tank as a function of time.
+            Positive is defined as a net mass flow rate entering the tank.
+        """
+        pass
+
+    @abstractmethod
+    def liquidVolume(self):
+        """Returns the volume of liquid inside the tank as a function
+        of time.
+
+        Parameters
+        ----------
+        time : float
+            Time in seconds.
+
+        Returns
+        -------
+        Function
+            Tank's liquid volume as a function of time.
+        """
+        pass
+
+    @functools.cached_property
+    def centerOfMass(self):
+        """Returns the center of mass of the tank's fluids as a function of
+        time.
+
+        Parameters
+        ----------
+        time : float
+            Time in seconds.
+
+        Returns
+        -------
+        Function
+            Center of mass of the tank's fluids as a function of time.
+        """
+
+        def centerOfMass(t):
+            self.evaluateTankState(t)
+
+            bottomCapMassBalance = (
+                self.bottomCapLiquidMass * self.bottomCap.filled_centroid
+                + self.bottomCapGasMass * self.bottomCap.empty_centroid
+            )
+            cylinderMassBalance = (
+                self.cylinderLiquidMass * self.cylinderLiquidCentroid
+                + self.cylinderGasMass * self.cylinderGasCentroid
+            )
+            upperCapMassBalance = (
+                self.upperCapLiquidMass * self.upperCapLiquidCentroid
+                + self.upperCapGasMass * self.upperCapGasCentroid
+            )
+
+            centerOfMass = (
+                bottomCapMassBalance + cylinderMassBalance + upperCapMassBalance
+            ) / (self.bottomCapMass + self.cylinderMass + self.upperCapMass)
+
+            return centerOfMass
+
+        centerOfMass = Function(
+            centerOfMass, inputs="Time (s)", outputs="Center of Mass (m)"
+        )
+
+        return centerOfMass
+
+    @functools.cached_property
+    def inertiaTensor(self):
+        """Returns the inertia tensor of the tank's fluids as a function of
+        time.
```

Comment:
TODO: make it clear relative to which origin and axes (coordinate system) is this being calculated.